gapis/api/vulkan/replay.go

```diff
@@ -90,6 +90,9 @@ func (a API) GetReplayPriority(ctx context.Context, i *device.Instance, h *captu
 				if devPhyInfo.GetDeviceId() != tracePhyInfo.GetDeviceId() {
 					continue
 				}
+				if devPhyInfo.GetDriverVersion() != tracePhyInfo.GetDriverVersion() {
```

Comment:
So when something fails we always return 1 here. Does this allow to return a proper error message to the user that's different when the device doesn't match and when the driver doesn't match?

Reply:
When a device is compatible we return 1, else we return 0 (the `return 1` is actually reached only if all checks pass). The semantic of GetReplayPriority is to return a priority which must be > 0 for the device to be considered valid.

This change does not enable us to return a precise error message for the reason why the device is not considered valid. The current architecture is such that for a given trace, the server returns the list of valid replay devices. Invalid replay devices are not listed, and so there is not even a chance to explain why they might be invalid. So the current architecture does not enable to say "we see you have a deviceABC plugged in, but it is not suitable for this trace". This limitation is tackled in "phase 2" of go/agi-replay-device.

The bug was originally filed because AGI would let the user try and replay on a device which had a different Vulkan driver version, this PR here disallow this, so I think it is a good first fix.

Reply:
You're right i wrongly read the code, thanks for the thorough explanation!